wandb/sdk/internal/sender.py

```diff
@@ -797,12 +825,12 @@ def send_metric(self, data: wandb_internal_pb2.Record) -> None:
             next_idx = len(self._config_metric_pbdict_list)
             self._config_metric_pbdict_list.append(md)
             self._config_metric_index_dict[metric.name] = next_idx
-        self._update_config()
+        self._update_config(data, debounce=True)
 
     def send_telemetry(self, data):
         telem = data.telemetry
         self._telemetry_obj.MergeFrom(telem)
-        self._update_config()
```

Comment:
We keep track of the latest telemetry in self._telemetry_obj
Hopefully this will move out of config so we would eventually call:
self._update_metadata()